rpcs3/xinput_pad_handler.cpp

```diff
@@ -2,12 +2,222 @@
+
+		// Triggers
+		values[xinput_pad_handler::XInputKeyCodes::LT] = state.Gamepad.bLeftTrigger;
+		values[xinput_pad_handler::XInputKeyCodes::RT]   = state.Gamepad.bRightTrigger;
+
+		// Sticks
+		int lx = state.Gamepad.sThumbLX;
+		int ly = state.Gamepad.sThumbLY;
+		int rx = state.Gamepad.sThumbRX;
+		int ry = state.Gamepad.sThumbRY;
+
+		// Left Stick X Axis
+		values[xinput_pad_handler::XInputKeyCodes::LSXNeg] = lx < 0 ? -lx - 1 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::LSXPos]   = lx > 0 ? lx : 0;
+
+		// Left Stick Y Axis
+		values[xinput_pad_handler::XInputKeyCodes::LSYNeg] = ly < 0 ? -ly - 1 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::LSYPos]   = ly > 0 ? ly : 0;
+
+		// Right Stick X Axis
+		values[xinput_pad_handler::XInputKeyCodes::RSXNeg] = rx < 0 ? -rx - 1 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::RSXPos]   = rx > 0 ? rx : 0;
+
+		// Right Stick Y Axis
+		values[xinput_pad_handler::XInputKeyCodes::RSYNeg] = ry < 0 ? -ry - 1 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::RSYPos]   = ry > 0 ? ry : 0;
+
+		// Buttons
+		const WORD buttons = state.Gamepad.wButtons;
+
+		// A, B, X, Y
+		values[xinput_pad_handler::XInputKeyCodes::A] = buttons & XINPUT_GAMEPAD_A ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::B] = buttons & XINPUT_GAMEPAD_B ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::X] = buttons & XINPUT_GAMEPAD_X ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::Y] = buttons & XINPUT_GAMEPAD_Y ? 255 : 0;
+
+		// D-Pad
+		values[xinput_pad_handler::XInputKeyCodes::Left]  = buttons & XINPUT_GAMEPAD_DPAD_LEFT ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::Right] = buttons & XINPUT_GAMEPAD_DPAD_RIGHT ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::Up]    = buttons & XINPUT_GAMEPAD_DPAD_UP ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::Down]  = buttons & XINPUT_GAMEPAD_DPAD_DOWN ? 255 : 0;
+
+		// LB, RB, LS, RS
+		values[xinput_pad_handler::XInputKeyCodes::LB] = buttons & XINPUT_GAMEPAD_LEFT_SHOULDER ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::RB] = buttons & XINPUT_GAMEPAD_RIGHT_SHOULDER ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::LS] = buttons & XINPUT_GAMEPAD_LEFT_THUMB ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::RS] = buttons & XINPUT_GAMEPAD_RIGHT_THUMB ? 255 : 0;
+
+		// Start, Back, Guide
+		values[xinput_pad_handler::XInputKeyCodes::Start] = buttons & XINPUT_GAMEPAD_START ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::Back]    = buttons & XINPUT_GAMEPAD_BACK ? 255 : 0;
+		values[xinput_pad_handler::XInputKeyCodes::Guide]   = buttons & GUIDE_BUTTON ? 255 : 0;
+
+		return values;
+	}
+};
+
+xinput_pad_handler::xinput_pad_handler(pad_handler handler)
+    : PadHandlerBase(handler)
+    , m_processor(makeProcessorFromType(handler))
+    , button_list(m_processor->GetButtonList())
 {
 	init_configs();
 
 	// Define border values
-	thumb_min = -32768;
+	thumb_min = -32767;
```

Comment:
are you sure about this?

Reply:
"Left thumbstick x-axis value. Each of the thumbstick axis members is a signed value between -32768 and 32767 describing the position of the thumbstick. A value of 0 is centered. Negative values signify down or to the left. Positive values signify up or to the right. "

Reply:
Yes, notice how this code here subtracts 1 to make it `32767` at most.
https://github.com/RPCS3/rpcs3/pull/6208/files/c8adbd50cae32a2798e4cc10ccb2b38792141001..42c54b86fcd3fc041692d6f004897d455ebc39fe#diff-37ad772e5aced056d3c766b7a498441dR167
This is not added by me, pre-PR code also did the same (just with `abs(X)`).

Reply:
then why did you change it,

Reply:
1 => (-x-1) = 0 => no input

Reply:
In XInput backend, `thumb_min` is used only in `NormalizeStickDeadzone` which in turn is operating on values from `GetButtonValues`. Due to this `- 1` subtraction present in the original (pre-PR) code, minimum value sticks can reach is `-(abs(-32768) - 1)`. Therefore, either `thumb_min = -32768` was wrong, **or** 1 should not be subtracted.

Reply:
So basing on what you said (which is totally true), `x - 1` logic is incorrect and should be removed, while `thumb_min = -32768;` was correct. Is that right?